### PR TITLE
Issue223 second solution: tiny update

### PR DESCRIPTION
**Pull request by josojo**
As Felix mentioned in https://github.com/gnosis/dex-contracts/pull/233, there is a little bit more gas efficient solution.

We save 350 gas per block of withdraw, by storing the batchIndex in the mapping hasCreditedIn, instead of storing the bool.

---
testplan: unit tests

### CODE REVIEW

File: contracts/EpochTokenLocker.sol [hunk 1]
```diff
@@ -36,7 +36,7 @@ contract EpochTokenLocker {
     mapping(address => mapping(address => BalanceState)) private balanceStates;
 
     // user => token => batchId => wasCredited
```
Comment (bh2smith) [inline contracts/EpochTokenLocker.sol]: I would think this comment should be adjusted to coincide with the changes to `lastCreditBatchId`.

**Review by bh2smith — CHANGES_REQUESTED** [review]
Looks great, just needs one adjustment in the comments.

**Comment by bh2smith** [comment]
> As Felix mentioned in #233

I am inconvenienced by these trails of links to things that I have to dig for and would appreciate if an, at least brief, description of the changes included here were provided.

**Comment by fleupold** [comment]
>I am inconvenienced by these trails of links

By clicking on the `x hours ago` on the comment you can get a better permalink that will take the reader straight to the comment at questions.

E.g. for this PR: https://github.com/gnosis/dex-contracts/pull/233#discussion_r340192911

**Review by fleupold — APPROVED** [review]
Interesting that it is 350 gas. I ran this example

```solidity
contract MapOfMaps {
    mapping(uint => mapping(uint => uint)) foo;
    mapping(uint => uint) bar;
    function fooo() public {
        foo[1][2] = 3;
    }
    function baar() public {
        bar[1] = 3;
    }
}
```

and got that bar is 100 gas cheaper than foo. So I guess you were right that it's not actually doing an SSTORE for each key but only saving some gas since it has to hash fewer words?

**Comment by josojo** [comment]
>and got that bar is 100 gas cheaper than foo. So I guess you were right that it's not actually doing an SSTORE for each key but only saving some gas since it has to hash fewer words?

But if I would be right, then my solution should have cost less gas, as we only store a bool instead of an int! 
However, maybe storing the bool is as expensive as storing the int, as we need to "start a new word" each time we store the bool.

Idk, I tested both version and Felix, your suggestions are definitively cheaper.

**Review by bh2smith — COMMENTED** [review]
Awesome!~

**Pull request merged**